components/server/src/workspace/workspace-starter.ts

```diff
@@ -603,12 +603,31 @@ export class WorkspaceStarter {
         if (WithEnvvarsContext.is(context)) {
             allEnvVars = allEnvVars.concat(context.envvars);
         }
-        const envvars = allEnvVars.map(uv => {
+
+        // we copy the envvars to a stable format so that things don't break when someone changes the
+        // EnvVarWithValue shape. The JSON.stringify(envvars) will be consumed by supervisor and we
+        // need to make sure we're speaking the same language.
+        const stableEnvvars = allEnvVars.map(e => { return { name: e.name, value: e.value }});
+
+        // we ship the user-specific env vars as OTS because they might contain secrets
+        const envvarOTSExpirationTime = new Date();
+        envvarOTSExpirationTime.setMinutes(envvarOTSExpirationTime.getMinutes() + 30);
+        const envvarOTS = await this.otsServer.serve(traceCtx, JSON.stringify(stableEnvvars), envvarOTSExpirationTime);
+
+        const envvars: EnvironmentVariable[] = [];
+        const ev = new EnvironmentVariable();
+        ev.setName("SUPERVISOR_ENVVAR_OTS");
+        ev.setValue(envvarOTS.token);
+        envvars.push(ev);
+
+        // TODO(cw): for the time being we're still pushing the env vars as we did before.
+        //           Once everything is running with the latest supervisor, we can stop doing that.
+        allEnvVars.forEach(e => {
             const ev = new EnvironmentVariable();
-            ev.setName(uv.name);
-            ev.setValue(uv.value);
-            return ev;
-        });
+            ev.setName(e.name);
+            ev.setValue(e.name);
```

Comment:
Oops:

<img width="331" alt="Screenshot 2022-02-08 at 10 15 40" src="https://user-images.githubusercontent.com/599268/152955580-ef1feaec-1e16-44b9-be48-ce04c431358f.png">

Reply:
Also sometimes the value seems to be `name + ' ' + value`? 🤔

<img width="389" alt="Screenshot 2022-02-08 at 10 17 08" src="https://user-images.githubusercontent.com/599268/152955949-56cf7b92-0ff1-4366-a1d5-9977ffbd4b93.png">